CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2021 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.19)
+
+project(greentea-client
+    VERSION 0.1
+    DESCRIPTION "Greentea client"
+    LANGUAGES C CXX
+)
+
+# Note: The INTERFACE type is currently required by Mbed OS
+# in order to inherit the platform's build options.
+add_library(greentea-client INTERFACE)
```

Comment:
Do we really need to force an INTERFACE library? The reason we do this in mbed-os is because of weak linking, which prevents us building normal static archives. With greentea-client we shouldn't have that issue.

Reply:
I initially thought compiler options were not passed properly by Mbed OS unless a library is INTERFACE. But if it's not the case, I'll switch it to a regular library.

Reply:
Changed